ReadMe.md

```diff
@@ -101,3 +107,57 @@ If you are still having the same error, then you can try by removing ccache. The
 ```
      $ apt-get remove ccache
 ```
+
+## Introduction to Envire and EnvireMars
+
+### Envire
+
+The envire graph is the backbone of the whole library. It stores arbitrary data and time & space transformations between the data. 
+
+Envire is mainly used in navigation as a package for expressing arbitrary information about a robot's environment, but it could be extended to also include robot environment interaction for manipulation.
+
+The reason for having a common datastructure for representing the environment is to simplify the interchange of algorithms between applications. Envire package is mainly made up of three types of objects. 
+
+1) Maps - give information in a given reference frame.
+
+2) Frames - provide the spatial context for the maps
+
+2) Operators - carry out operations on maps by considering the maps' frame associations 
+
+#### More details 
+
+0. [Official introduction to Envire](http://envire.github.io/docs/intro_about_envire.html)
+
+0. [Hidalgo Carrió, et al, EnviRe - Environment Representation for Long-term Autonomy, AI for Long-term Autonomy Workshop on the International Conference on Robotics and Automation (ICRA). IEEE International Conference on Robotics and Automation (ICRA-16), May 16-20, Stockholm, Sweden, 2016.](https://www.dfki.de/fileadmin/user_upload/import/8365_Hidalgo-Carrio_et_al._-_2016_-_EnviRe_-_Environment_Representation_for_Long_Term_Autonomy.pdf)
+
+0. [Official introduction to Envire_Core](https://github.com/envire/envire-envire_core#build-and-test-status)
+
+### MARS 
+
+MARS (Machina Arte Robotum Simulans) is a cross-platform simulation and visualisation tool created for robotics research developed at the DFKI-RIC and the University of Bremen. It runs on (Ubuntu) Linux, Mac and Windows and consists of a core framework containing all main simulation components.
+
+Those Components are: 
+
+1) Open Scene Graph (OSG) for its visualisation
+2) Open Dynamics Engine (ODE) for the physical simulation
+3) Qt for the GUI
+
+### UML Diagrams of EnvireMars
+
+1) [Package Diagram of EnvireMars](https://github.com/HasithDasa/simulation-buildconf/blob/Doc/diagrams/Package_Diagram.svg)
+2) [Activity Diagram of EnvireMars](https://github.com/HasithDasa/simulation-buildconf/blob/Doc/diagrams/EnvireMars_Activity_Diagram.svg)
+
+#### More details  
```

Comment:
Move these More details to the mars subsection, before the subsection of the UML diagrams.